### ISSUE TITLE
facebook ads extension

### ISSUE THREAD

**Issue by doandbe** [facebookincubator/Facebook-For-OpenCart#299]
I have installed Facebook ads extension automatically from Facebook business for some reason it created a second account with the same pixel number on Facebook business manager.
Although on opencart it looks connected to the pixel and on Facebook business manager too when i try to set up new events from business manager to my website it says that there is no pixel connected to this account.
i have also installed messenger from Facebook ads extension which also doesn't work and doesn't show on my website
journal 3.0.40 Opencart Version: 2.3.0.2
web site https://www.doandbe.gr/

![add events](https://user-images.githubusercontent.com/58469642/70047659-20270e00-15d1-11ea-990e-2a10a7b3acd7.PNG)
![pixel 1](https://user-images.githubusercontent.com/58469642/70047662-20bfa480-15d1-11ea-9ed9-4a348afb67e7.PNG)
![pixel 2](https://user-images.githubusercontent.com/58469642/70047664-20bfa480-15d1-11ea-9864-35910fc8ef64.PNG)

**Comment by shue249** [comment]
@doandbe 

We looked at your website and believe that the setup for the plugin is not complete.

To complete the setup, you can refer to this guide and the video.
https://github.com/facebookincubator/Facebook-For-OpenCart/blob/master/INSTALL_GUIDE.md#setup-for-facebook-pixel-catalog-and-page-shop

https://drive.google.com/open?id=1JPXQ0mS1pGk2Bat9RV02vYNdbRfCLP9r

Let us know if this helps you.

**Comment by shue249** [comment]
@doandbe 

Another suggestion. We found that you are still using an earlier version of the plugin - 2.1.3. We have the latest version 2.1.13 and it is available on here - https://github.com/facebookincubator/Facebook-For-OpenCart/releases/tag/v2.1.13

Can you remove away the older version and install the newer version of the plugin and see if the pixel firing is now working.

**Comment by doandbe** [comment]
hello i am using seo mega kit plus, i tried to to find out and disable cache but i couldnt find it my oc v 2.3.0.2 and it doesnt show like on your picture i installed new version of facebook manager and erased the previous one. I found out that using google chrome with pixel helper let me set up new events and i did becasue i see it is working,the problem was aslso the mozilla firefox,installing the new version created messenger chat icon on my site but i can see it only with google chrome weird!!!
but with the new instllation fixed one problem created another 2. 1 when i click on facebook module is shows all the time checking for upload status,if i try to refresh catalalog and fetch products for catalog it takes ages, ans last but not least it shows this message when i open back end page " There is an error making API calls using the access token. The error encountered is "

```
> Error validating access token: The session has been invalidated because the user changed their password or Facebook has changed the session for security reasons.".
> To update your access token, please click on Facebook Business Extension, Manage Settings, go to Advanced options and click on Update token.
```
of course on advanced options there is update token

please help with these issues

**Comment by shue249** [comment]
@doandbe Thank you for the feedback.

1. We done a quick check on the pixel setup on your website. The events are now firing correctly on your product page for ViewContent and AddToCart. If you like, you can try to make a mock purchase on your website to verify that the Purchase event is also correctly fired.
![image](https://user-images.githubusercontent.com/11957585/70131598-8be6a500-16bd-11ea-83b0-ec5326c08dff.png)

2. For the Messenger chat plugin, a quick test on Firefox and Chrome from our side shows that it is working.
![image](https://user-images.githubusercontent.com/11957585/70131900-0b747400-16be-11ea-8cfc-ce52638c6fec.png)

3. For the error message on the access token, we are now working on a fix and we are targeting to roll out the fix very soon. Once the fix is ready, we will update you soonest.

**Comment by doandbe** [comment]
thanks a lot you were very helpful , just one last thing as it regards "when i click on facebook module is shows all the time checking for upload status,if i try to refresh catalalog and fetch products for catalog it takes ages" is has something to do with the token,if no how can i fix it

**Comment by shue249** [comment]
@doandbe The fix that we are working on now will solve the catalog refresh issue that you are facing.

**Comment by doandbe** [comment]
ok thanks a lot

**Comment by shue249** [comment]
@doandbe  

We have released a new version of the plugin, 2.2.0. In this new version, we have changed the catalog sync to using 100% feed file and the API/access token is no longer needed. This will resolve your current problem. You need to do an upgrade of the existing catalog and the steps to do that is highlighted in the release notes.

Here is the release notes:
2.2.0 - 11 Nov 2019
* Changed the product catalog sync method. On earlier versions, the plugin adopted 2 approaches - using feed file for initial catalog sync and API for subsequent product changes. From this version onwards, the plugin is adopting feed file for both initial catalog sync as well as subsequent product changes.
  1. There will be a daily feed pull to update the product catalog changes from your OpenCart website to Facebook product catalog.
  2. For existing plugin merchants, you need to perform a step to automatically generate the daily pull mechanism. Access the Facebook Business Extension module, click on Manage Settings, click on Advanced options. You will be able to see the notification indicating that the Catalog sync is successful. See screenshot here - https://drive.google.com/open?id=1dAF88Spg-iAwPXmkwcVvAqg82UYJ5IJ0
  3. Access token is no longer needed and there is no longer any need to update the access token nor to keep the Facebook account signed in.
  4. Better integration with other 3rd party product plugins. You can use any of the 3rd party product plugins to perform add/edit/delete product changes. These product changes will be automatically updated to Facebook via the daily feed mechanism.
  5. You can trigger immediate product sync to Facebook. Access the Facebook Business Extension module, click on Manage Settings, click on Your Products. Click on Fetch Now to sync your products to Facebook. See screenshot here - https://drive.google.com/open?id=1bCTT4gU4gC8pi6T6HLhpe2fkHWI-RvhN

We will be closing out this ticket, feel free to reach to us on this ticket if you have any further questions - https://github.com/facebookincubator/Facebook-For-OpenCart/issues/302
